tests/sentry/integrations/source_code_management/test_commit_context_slo.py

```diff
@@ -95,3 +95,32 @@ def test_get_commit_context_all_frames(self, mock_record):
         assert result == []
         assert len(mock_record.mock_calls) == 2
         assert_slo_metric(mock_record, EventLifecycleOutcome.SUCCESS)
+
+    @patch("sentry.integrations.utils.metrics.EventLifecycle.record_event")
+    def test_get_blame_for_files_gitlab_halt(self, mock_record):
```

Comment:
codecov is complaining there's no test for gitlab recording a failure and no test for _default_lifecycle_handler